src/coq_elpi_builtins.ml

```diff
@@ -534,11 +539,11 @@ let is_global_level env u =
 [%%endif]
 
 let err_if_contains_alg_univ ~depth t =
-  let global_univs = UGraph.domain (Environ.universes (Global.env ())) in
+  let env = Global.env () in
   let is_global u = 
     match Univ.Universe.level u with
     | None -> true
-    | Some l -> Univ.Level.Set.mem l global_univs in
+    | Some l -> is_global_level env l in
```

Comment:
you could merge this in master separately btw

Reply:
right, but I hope to merge this soon, keeping the PR going is draining too much of my time